storm-core/src/jvm/org/apache/storm/daemon/supervisor/SyncSupervisorEvent.java

```diff
@@ -0,0 +1,626 @@
+
+            for (Map.Entry<Integer, LocalAssignment> entry : allAssignment.entrySet()) {
+                if (supervisorData.getiSupervisor().confirmAssigned(entry.getKey())) {
+                    newAssignment.put(entry.getKey(), entry.getValue());
+                    assignedStormIds.add(entry.getValue().get_topology_id());
+                }
+            }
+
+            Set<String> srashStormIds = verifyDownloadedFiles(conf, supervisorData.getLocalizer(), assignedStormIds, allDownloadedTopologyIds);
+            Set<String> downloadedStormIds = new HashSet<>();
+            downloadedStormIds.addAll(allDownloadedTopologyIds);
+            downloadedStormIds.removeAll(srashStormIds);
+
+            LOG.debug("Synchronizing supervisor");
+            LOG.debug("Storm code map: {}", stormcodeMap);
+            LOG.debug("All assignment: {}", allAssignment);
+            LOG.debug("New assignment: {}", newAssignment);
+            LOG.debug("Assigned Storm Ids {}", assignedStormIds);
+            LOG.debug("All Downloaded Ids {}", allDownloadedTopologyIds);
+            LOG.debug("Checked Downloaded Ids {}", srashStormIds);
+            LOG.debug("Downloaded Ids {}", downloadedStormIds);
+            LOG.debug("Storm Ids Profiler Actions {}", stormIdToProfilerActions);
+
+            // download code first
+            // This might take awhile
+            // - should this be done separately from usual monitoring?
+            // should we only download when topology is assigned to this supervisor?
+            for (Map.Entry<String, String> entry : stormcodeMap.entrySet()) {
+                String stormId = entry.getKey();
+                if (!downloadedStormIds.contains(stormId) && assignedStormIds.contains(stormId)) {
+                    LOG.info("Downloading code for storm id {}.", stormId);
+                    try {
+                        downloadStormCode(conf, stormId, entry.getValue(), supervisorData.getLocalizer());
+                    } catch (Exception e) {
+                        if (Utils.exceptionCauseIsInstanceOf(NimbusLeaderNotFoundException.class, e)) {
+                            LOG.warn("Nimbus leader was not available.", e);
+                        } else if (Utils.exceptionCauseIsInstanceOf(TTransportException.class, e)) {
+                            LOG.warn("There was a connection problem with nimbus.", e);
+                        } else {
+                            throw e;
+                        }
+                    }
+                    LOG.info("Finished downloading code for storm id {}", stormId);
+                }
+            }
+
+            LOG.debug("Writing new assignment {}", newAssignment);
+
+            Set<Integer> killWorkers = new HashSet<>();
+            killWorkers.addAll(existingAssignment.keySet());
+            killWorkers.removeAll(newAssignment.keySet());
+            for (Integer port : killWorkers) {
+                supervisorData.getiSupervisor().killedWorker(port);
```

Comment:
This place just info ISupervisor killedWorker(port);

Don't do supervisorData.getWorkerManager.shutdown() worker?